tools/cli/src/mentat_cli/repl.rs

```diff
@@ -7,63 +7,103 @@
+// TODO: Should this actually reflect the current open brace?
 const MORE_PROMPT: &'static str = "mentat.> ";
-/// Prompt when a `.block` command is in effect
-const BLOCK_PROMPT: &'static str = "mentat+> ";
+
+lazy_static! {
+    static ref COMMAND_HELP: HashMap<&'static str, &'static str> = {
+        let mut map = HashMap::new();
+        map.insert(HELP_COMMAND, "Show help for commands.");
+        map.insert(OPEN_COMMAND, "Open a database at path.");
+        map
+    };
+}
 
 /// Executes input and maintains state of persistent items.
 pub struct Repl {
+    store: Store,
 }
 
 impl Repl {
     /// Constructs a new `Repl`.
-    pub fn new() -> Repl {
-        Repl{}
+    pub fn new(db_name: Option<String>) -> Repl {
+        Repl{
+            store: Store::new(db_name),
+        }
     }
 
-
     /// Runs the REPL interactively.
     pub fn run(&mut self) {
-        let mut more = false;
+        let mut more: Option<Command> = None;
         let mut input = InputReader::new();
 
         loop {
-            let res = input.read_input(if more { MORE_PROMPT } else { DEFAULT_PROMPT });
-        //     let res = if self.read_block {
-        //         self.read_block = false;
-        //         input.read_block_input(BLOCK_PROMPT)
-        //     } else {
-        //         input.read_input(if more { MORE_PROMPT } else { DEFAULT_PROMPT })
-        //     };
+            let res = input.read_input(if more.is_some() { MORE_PROMPT } else { DEFAULT_PROMPT });
 
             match res {
-                Command(name, args) => {
-                    debug!("read command: {} {:?}", name, args);
-
-                    more = false;
-                    self.handle_command(name, args);
+                MetaCommand(cmd) => {
+                    debug!("read command: {:?}", cmd);
+                    more = None;
+                    self.handle_command(cmd);
                 },
                 Empty => (),
-                More => { more = true; },
+                More(cmd) => { more = Some(cmd); },
                 Eof => {
                     if input.is_tty() {
                         println!("");
                     }
                     break;
-                }
+                },
+                InputError(err) => {
+                    println!("{}", err);
+                    more = None;
+                },
             };
         }
     }
 
     /// Runs a single command input.
-    fn handle_command(&mut self, cmd: String, args: Option<String>) {
-        println!("{:?} {:?}", cmd, args);
+    fn handle_command(&mut self, cmd: Command) {
+        match cmd {
+            Command::Help(args) => self.help_command(args),
+            Command::Open(db) => {
+                self.store.open(Some(db));
+            },
+            Command::Close => self.store.close(),
+            Command::Err(message) => println!("{}", message),
+            _ => unimplemented!(),
+        }
+    }
+
+    fn help_command(&self, args: Vec<String>) {
```

Comment:
Github wouldn't let me comment on the `tools/cli/src/mentat_cli/store.rs` file, so here it is:

I expect we will want to grow this into something (like `Conn`) in `core`, so rather than `expect()` and swallowing the underlying error, can we return `Result<Store, AppropriateErrorType>` and handle errors like this at a higher level?

If you really want `Store` to be CLI specific and to handle errors inline, I think you need to surface the underlying error in some way.